### DISCUSSION TITLE
Release 2.0.0b3 - for testing only

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#7070] [General]
borgbackup 2.0.0b3 beta release - do not use this on production backup repositories.

Long changelog:
https://borgbackup.readthedocs.io/en/2.0.0b3/changes.html#version-2-0-0b3-2022-10-02

If you use pip to install this, use: pip install "borgbackup==2.0.0b3"

Docs: https://borgbackup.readthedocs.io/en/2.0.0b3/

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/2.0.0b3'>Release 2.0.0b3 - for testing only</a>.</em>

**Comment by real-yfprojects** [comment]
There is no documentation for `rcompress` yet.
Is it useful for reducing the disk space of a repository?

**Reply by ThomasWaldmann** [reply]
https://borgbackup.readthedocs.io/en/2.0.0b3/usage/rcompress.html
or
```
borg rcompress --help
```

The docs building had some hickup, but they are available at https://borgbackup.readthedocs.io/en/2.0.0b3/ now (and I'll add rcompress to the index).

**Comment by m3nu** [comment]
Also updated this new beta version on all [BorgBase.com](https://www.borgbase.com/) repos that have their version set to `2.0 Beta` under *Advanced > BorgBackup Version*.

**Reply by ThomasWaldmann** [reply]
Users should please recreate their borg2 repos - the repo format between betas is not compatible and there is no upgrade procedure.

**Comment by real-yfprojects** [comment]
What is the fastest way to determine the version/compatibility of a repository?

**Reply by ThomasWaldmann** [reply]
just create a new repo for each beta.

**Reply by real-yfprojects** [reply]
For vorta dev we would like to determine whether a repository is incompatible with borg v2 in case _v2_ is installed.

**Reply by ThomasWaldmann** [reply]
for borg 1.x, the repo version will be 1, for borg 2 it will be 2 (see repo config).

you can use this to make the decision when borg 2 is release (or close to release), but you can not determine compatibility between borg 2 betas using this method obviously, so just assume that the betas are not compatible to each other.

**Reply by real-yfprojects** [reply]
> for borg 1.x, the repo version will be 1, for borg 2 it will be 2 (see repo config).

Thanks. So this will only be available for local repos, I guess.

**Reply by ThomasWaldmann** [reply]
For the repo URL in borg 1 as well as for `--repo` in borg 2, the client will reject talking to a wrong version remote repo.

For borg 2 and `--other-repo`, borg 2.0 will be able to talk to a borg 1 repo to use `borg create/transfer`.

**Comment by dietmargoldbeck** [comment]
prune with pattern not working?

With the new beta 2.0.0b3 prune does not work anymore with a pattern. If i do *not* use a pattern it works normally.
If i use e.g. -a 'b*' it does not match any backups although all archives start with b

Has the usage changed? I only noted the change of the long option name

**Reply by infectormp** [reply]
> the --glob-archives option was renamed to --match-archives (the short option name -a is unchanged) and extended to support different pattern styles:
> 
> id: for identical string match (this is the new default!)
> sh: for shell pattern / globbing match (this was used by --glob-archives)
> re: for regular expression match
> So you might need to edit your scripts like e.g.:
> 
> borg 1.x: --glob-archives 'myserver-*'
> borg 2.0: --match-archives 'sh:myserver-*'

I suppose you need to use `-a 'sh:b*'`

**Reply by ThomasWaldmann** [reply]
btw, such stuff can be also read in `docs/changes.rst` (you should read that anyway when testing betas or doing upgrades).

**Comment by dietmargoldbeck** [comment]
git check shows some orphans

While testing 2.0.0b3 i noticed that borg check reports  a small number of orphans. They can be increased
with a backup which gets interrupted by a network problem giving a checkpoint archive.
(it seems like 3 orphans per run with checkpoint, but not sure. I used borg 2 in a train over LTE to backup my
notebook)

Should i run borg check --repair?
Any debugging recommendations?

**Reply by ThomasWaldmann** [reply]
Yes, guess `borg check --repair` would fix it. Orphans (present, but unreferenced chunks) are a rather harmless problem. The opposite case would be worse.

**Comment by m3nu** [comment]
With Python 3.11 released today, has anyone done speed tests, if this benefits Borg v2? And what's the expected speedup between v1 and v2?

If not, I may do this as my own little project.

**Reply by ThomasWaldmann** [reply]
I didn't do speed tests, but it would be interesting to see the differences between 3.9/3.10/3.11.

Also, building the binary with pyinstaller and py311 would be interesting.

**Reply by m3nu** [reply]
> would be interesting to see the differences between 3.9/3.10/3.11

Here some simple benchmarks using `borg benchmark crud`. Run on an idle Macbook Pro 2017.

https://github.com/m3nu/borg-benchmark

For Python 3.9 vs 3.11: A bit faster for most metrics. 10% maybe?

For Borg 1.2 vs 2 (beta3): About 20% faster on most CREATES and 3-5x faster for most READS. Slower on DELETES for some reason.

**Reply by ThomasWaldmann** [reply]
Could you add the unit to the xlsx table (so one can know what the numbers mean) and maybe export it to pdf?

**Reply by m3nu** [reply]
Sure. Done.

**Comment by mbo77** [comment]
I'm testing with the current beta version and it seems I can't transfer a local unencrypted 1.2 repository via transfer:

```
Local Exception
Traceback (most recent call last):
  File "/usr/local/lib64/python3.10/site-packages/borg/archiver/__init__.py", line 594, in main
    exit_code = archiver.run(args)
  File "/usr/local/lib64/python3.10/site-packages/borg/archiver/__init__.py", line 522, in run
    return set_ec(func(args))
  File "/usr/local/lib64/python3.10/site-packages/borg/archiver/_common.py", line 223, in wrapper
    return method(self, args, **kwargs)
  File "/usr/local/lib64/python3.10/site-packages/borg/archiver/_common.py", line 155, in wrapper
    return method(self, args, repository=repository, cache=cache_, **kwargs)
  File "/usr/local/lib64/python3.10/site-packages/borg/archiver/transfer_cmd.py", line 66, in do_transfer
    for item in other_archive.iter_items():
  File "/usr/local/lib64/python3.10/site-packages/borg/archive.py", line 610, in iter_items
    for item in self.pipeline.unpack_many(
  File "/usr/local/lib64/python3.10/site-packages/borg/archive.py", line 282, in unpack_many
    for data in self.fetch_many(ids):
  File "/usr/local/lib64/python3.10/site-packages/borg/archive.py", line 313, in fetch_many
    _, data = self.repo_objs.parse(id_, cdata)
  File "/usr/local/lib64/python3.10/site-packages/borg/repoobj.py", line 157, in parse
    meta["csize"] = len(data_compressed)
TypeError: 'NoneType' object does not support item assignment

Platform: Linux ds9 6.0.8-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Nov 11 15:03:58 UTC 2022 x86_64
Linux: Unknown Linux
Borg: 2.0.0b3  Python: CPython 3.10.8 msgpack: 1.0.3 fuse: llfuse 1.4.2 [pyfuse3,llfuse]
PID: 3759256  CWD: /datapool
sys.argv: ['/usr/local/bin/borg', 'transfer', '--other-repo', '/mnt/backup/bDISK/borg', '--repo', '/datapool/borg2', '--upgrader=From12To20', '--dry-run']
SSH_ORIGINAL_COMMAND: None
```
Installed beta version via pip.

**Reply by ThomasWaldmann** [reply]
@mbo77 see there: https://github.com/borgbackup/borg/issues/7156

**Reply by mbo77** [reply]
Looks good. Dry-run started now. Thank you.
